### PR TITLE
config: use format_str() instead of hand-rolled snprintf()

### PR DESCRIPTION
**Pull request by emersion**

### CODE REVIEW

File: sway/config.c [hunk 1]
```diff
@@ -368,10 +362,7 @@ static char *get_config_path(void) {
 
 	const char *config_home = getenv("XDG_CONFIG_HOME");
 	if ((config_home == NULL || config_home[0] == '\0') && home != NULL) {
-		size_t size_fallback = 1 + strlen(home) + strlen("/.config");
-		config_home_fallback = calloc(size_fallback, sizeof(char));
-		if (config_home_fallback != NULL)
-			snprintf(config_home_fallback, size_fallback, "%s/.config", home);
+		config_home_fallback = format_str("%s/.config", home);
 		config_home = config_home_fallback;
```
Comment (bl4ckb0ne) [inline sway/config.c:366]: `config_home_fallback` could go

```
config_home = format_str("%s/.config", home);
```

Comment (emersion) [reply sway/config.c:366]: Unfortunately no, we still need to free that string.

Comment (bl4ckb0ne) [reply sway/config.c:366]: Indeed, forgot about that.

**Review by bl4ckb0ne — APPROVED** [review]
LGTM!

**Pull request merged**